### PR TITLE
correct a metric about compaction filter

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What is changed and how it works?
Issue Number: Close #xxx

What's Changed:

```commit-message
correct a metric about compaction filter
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- PR to update `pingcap/tidb-ansible`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/server/gc_worker/gc_worker.rs [hunk 1]
```diff
@@ -364,15 +364,16 @@ where
         };
         let mut gc_info = GcInfo::default();
         let mut next_gc_key = keys.next();
+        let mut handled_keys = 0;
         while let Some(ref key) = next_gc_key {
-            GC_COMPACTION_FILTER_MVCC_DELETION_HANDLED.inc();
             if let Err(e) = self.gc_key(safe_point, key, &mut gc_info, &mut txn, &mut reader) {
                 error!(?e; "GC meets failure"; "key" => %key,);
```
Comment (Connor1996) [inline src/server/gc_worker/gc_worker.rs:370]: add fail metrics?

File: src/server/gc_worker/gc_worker.rs [hunk 2]
```diff
@@ -364,15 +364,16 @@ where
...
                 // Switch to the next key if meets failure.
                 gc_info.is_completed = true;
             }
 
             if gc_info.is_completed {
+                handled_keys += 1;
```
Comment (MyonKeminta) [inline src/server/gc_worker/gc_worker.rs:376]: Why not `inc` directly here?

Comment (hicqu) [reply src/server/gc_worker/gc_worker.rs:376]: `GC_COMPACTION_FILTER_MVCC_DELETION_HANDLED` is only for mvcc deletions come from compaction filter.

**Review by Connor1996 — APPROVED** [review]
LGTM

**Review by MyonKeminta — APPROVED** [review]

**Comment by hicqu** [comment]
图片
Here is the modified metric panel.

**Pull request merged**